sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -2251,7 +2251,11 @@ class DatasetLargeResultCollectingSuite extends QueryTest
   with SharedSparkSession {
 
   override protected def sparkConf: SparkConf = super.sparkConf.set(MAX_RESULT_SIZE.key, "4g")
-  test("collect data with single partition larger than 2GB bytes array limit") {
+  // SPARK-41193: Ignore this suite because it cannot run successfully with Spark
+  // default Java Options, if user need do local test, please make the following changes:
+  // - Maven test: change `-Xmx4g` of `scalatest-maven-plugin` in `sql/core/pom.xml` to `-Xmx10g`
+  // - SBT test: change `-Xmx4g` of `Test / javaOptions` in `SparkBuild.scala` to `-Xmx10g`
+  ignore("collect data with single partition larger than 2GB bytes array limit") {
```

Comment:
@liuzqt, I know this was iterated on multiple times to get it to work - instead of the shared local spark session, did it work locally when using a local spark cluster instead ?

Reply:
Yes @LuciferYang  is right, need to change `-Xmx4g` to `-Xmx10g` to make it work (it works for both shared local session and local cluster, but without the change neither work).  

Thanks for the fix!  Previously I only tested this using IDE and I guess it increased the mem under the hood......Sorry for the inconvenience.

Reply:
So how do we move forward? This is a blocking for developers

Reply:
I think we can leave it as ignore for now with the comments about using larger mem to make it work. I'm not sure if we're able to configure the build args for a specific test suite.